ops/model.py

```diff
@@ -1937,8 +1940,22 @@ def from_name(cls, name: str, message: str):
             return cls._statuses[typing.cast(StatusName, name)](message)
 
     @classmethod
-    def register(cls, child: Type['StatusBase']):
-        """Register a Status for the child's name."""
+    def register(cls, child: Type[_StatusClass]) -> Type[_StatusClass]:
+        """Class decorator to register a subclass for lookup using :meth:`from_name`.
+
+        Note: this method is intended for internal use only.
+        It is used to make the six valid Juju statuses available by name.
+
+        .. code-block:: python
+
+            _StatusClass = typing.TypeVar('_StatusClass', bound='StatusBase')
```

Comment:
We have a few other places where we have types referred to in the docs but that aren't in the docs (_ModelBackend, _ModelCache, etc). I think this is like that, in that we don't expect people to be using this so providing the exact type in the docs isn't that important. So I would personally leave this bit out and just have the `_StatusClass` not linked or defined in the docs.